requirements.txt

```diff
@@ -1,10 +1,8 @@
-click==7.1.2
+docopt
 Flask==1.1.2
-itsdangerous==1.1.0
 Jinja2==2.11.2
 MarkupSafe==1.1.1
 pyyaml==5.3.1
 requests==2.25.1
 ruamel.yaml==0.16.12
-ruamel.yaml.clib==0.2.2
-Werkzeug==1.0.1
```

Comment:
How did you generate this? Pretty darn sure `Werkzeug` is needed.

Reply:
Werkzeug is pulled in as part of another package. Do we need to lock it?

Reply:
Ah ok, if flask is pulling it in, all good. Still curious how you generated the new requirements though. `pip freeze > requirements.txt` is the only way I know, but pretty sure that adds everything installed in your venv, whether it's used in the code or not.

Reply:
Yep, that's exactly correct!

I was wondering why all this was in here, I thought maybe you had made use of `pip-tools` or something.

Instead of capturing the environment, I just add the libs I want/need and let the `pip` tool sort out dependencies. When I get bitten by version issues I tend to lock down versions only at that time. This might be a bit cavalier of me though.

Reply:
This way the `Werkzeug` lib will come in at the latest non-breaking version that is depended on by `Flask`, which I think is better, and not as flippant as I would have it?

Reply:
I do prefer to lock all versions, otherwise shit can break suddenly when you build, and you'll probably spend some time figuring out it wasn't your code changes that broke the shit.